packages/discovery-provider/src/queries/get_comments.py

```diff
@@ -397,6 +397,141 @@ def filter_mentions(mention):
+        args, default_limit=COMMENT_ROOT_DEFAULT_LIMIT
+    )
+
+    user_comments = []
+    db = get_db_read_replica()
+
+    with db.scoped_session() as session:
+        mentioned_users = (
+            session.query(
+                CommentMention.comment_id,
+                CommentMention.user_id,
+                User.handle,
+                CommentMention.is_delete,
+            )
+            .join(User, CommentMention.user_id == User.user_id)
+            .subquery()
+        )
+
+        react_count_subquery = (
+            session.query(
+                CommentReaction.comment_id,
+                func.count(CommentReaction.comment_id).label("react_count"),
+            )
+            .filter(CommentReaction.is_delete == False)
+            .group_by(CommentReaction.comment_id)
+            .subquery()
+        )
+
+        user_comments = (
+            session.query(
+                Comment,
+                func.coalesce(react_count_subquery.c.react_count, 0).label(
+                    "react_count"
+                ),
+                CommentNotificationSetting.is_muted,
+                func.array_agg(
+                    func.json_build_object(
+                        "user_id",
+                        mentioned_users.c.user_id,
+                        "handle",
+                        mentioned_users.c.handle,
+                        "is_delete",
+                        mentioned_users.c.is_delete,
+                    )
+                ).label("mentions"),
+                Track.owner_id.label("artist_id"),
+            )
+            .outerjoin(
+                Track,
+                Comment.entity_id == Track.track_id,
+            )
+            .outerjoin(
+                mentioned_users,
+                Comment.comment_id == mentioned_users.c.comment_id,
+            )
+            .outerjoin(
+                react_count_subquery,
+                Comment.comment_id == react_count_subquery.c.comment_id,
+            )
+            .outerjoin(
+                CommentNotificationSetting,
+                (Comment.comment_id == CommentNotificationSetting.entity_id)
+                & (CommentNotificationSetting.entity_type == "Comment"),
+            )
+            .group_by(
+                Comment.comment_id,
+                react_count_subquery.c.react_count,
+                CommentNotificationSetting.is_muted,
+                Track.owner_id,
+            )
+            .filter(
+                Comment.user_id == user_id,
+                Comment.is_delete == False,
+                Comment.is_visible == True,
+                Comment.entity_type == "Track",
+            )
+            .order_by(desc(Comment.created_at))
+            .offset(offset)
+            .limit(limit)
+        ).all()
+
+        user_comments_res = []
+        for [user_comment, react_count, is_muted, mentions, artist_id] in user_comments:
+            replies = get_replies(
```

Comment:
Do we actually need the replies? I didn't think the UI we were showing would include these and I imagine it probably adds a lot of overhead to the call.

Reply:
I wanted to use the same base comment type on the way back and figure this was futureproofing, but it's probably not worth it for now yea